src/DynamoPackages/PackageLoader.cs

```diff
@@ -399,6 +440,78 @@ public Package ScanPackageDirectory(string directory)
+                        discoveredPkg.Name, discoveredPkg.RootDirectory));
+            }
+
+            foreach (var manifestFile in discoveredPkg.Header.node_libraries)
+            {
+
+                //Try to get the assembly name from the manifest file
+                string filename;
+                try
+                {
+                    filename = new AssemblyName(manifestFile).Name + ".dll";
+                }
+                catch
+                {
+                    throw new LibraryLoadFailedException(directory,
+                        String.Format(
+                            "A package called {0} found at {1} has improperly defined its node libraries in the package manifest.  Ignoring it.",
+                            discoveredPkg.Name, discoveredPkg.RootDirectory));
+                }
+
+
+                //Verify the node library exists in the package bin directory
+                var filepath = Path.Combine(discoveredPkg.BinaryDirectory, filename + ".dll");
+                if (!File.Exists(filepath))
+                {
+                    throw new LibraryLoadFailedException(directory,
+                        String.Format(
+                            "A package called {0} found at {1} is missing dlls which are defined in the package manifest.  Ignoring it.",
+                            discoveredPkg.Name, discoveredPkg.RootDirectory));
+                }
+
+                //Verify that you can load the node library assembly into a Reflection only context
+                Assembly asm;
+                try
+                {
+                    asm = Assembly.ReflectionOnlyLoadFrom(filename);
+                }
+                catch
+                {
+                    throw new LibraryLoadFailedException(directory,
+                        String.Format(
+                            "A package called {0} found at {1} has dlls defined in the package manifest which could not be loaded.  Ignoring it.",
+                            discoveredPkg.Name, discoveredPkg.RootDirectory));
+                }
+
+                //Verify teh node libarary has a verified signed certificate
+                var cert = asm.Modules.FirstOrDefault()?.GetSignerCertificate();
+                if (cert != null)
+                {
+                    var cert2 = new System.Security.Cryptography.X509Certificates.X509Certificate2(cert);
+                    if (cert2.Verify())
+                    {
+                        continue;
+                    }
+                }
+
+                throw new LibraryLoadFailedException(directory,
+                    String.Format("A package called {0} found at {1} did not have signed dll files.  Ignoring it.",
+                        discoveredPkg.Name, discoveredPkg.RootDirectory));
+            }
+
+            discoveredPkg.HasSignedEntryPoints = true;
```

Comment:
what conditions will make this true?

Reply:
You get here when you have looped through each file in the node_libraries and verified it has a cert that is verified.  Everywhere else you hit a exception